### PR TITLE
SCC-1306 | BUG - "Check Your Details"

### PR DESCRIPTION
**Pull request by Neozxc**

### CODE REVIEW

**Review by ccatosdevelopment — APPROVED** [review]

File: app/pages/answers/changeaddress/RemovePostalAddressPage.scala [hunk 1]
```diff
@@ -16,14 +16,33 @@
...
+import scala.util.Try
+
 case object RemovePostalAddressPage extends QuestionPage[RemovePostalAddress] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "removePostalAddress"
 
+  override def cleanup(value: Option[RemovePostalAddress], userAnswers: UserData): Try[UserData] =
+    (value, userAnswers.get(RemovePostalAddressPage)) match {
```
Comment (owen-opencast) [inline app/pages/answers/changeaddress/RemovePostalAddressPage.scala:34]: If you've answered Somewhere else, then click "change", then click submit without changing the answer, the data is wiped and the user must complete the journey again. Instead, the user to return to Check Your Answers.

cleanup is run _after_ the value is set in user answers, so userAnswers.get(RemovePostalAddressPage) will always match Some(_). Have a look at IsNewAddressInUkPage - the way to do it is to figure out the users previous answer with other answers...

- If they've previously answered "Main", there should be no further answers... i.e. EnterYourUkAddressPage should return None.
- If they've previously answered "Somewhere else", then there should be an answer somewhere. 

You could maybe check with !AddressJourney.hasAnsweredUk && !AddressJourney.hasAnsweredInternational - i.e. no answers = main.

**Review by owen-opencast — CHANGES_REQUESTED** [review]

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

File: app/pages/answers/changeaddress/RemovePostalAddressPage.scala [hunk 2]
```diff
@@ -16,14 +16,38 @@
...
+    value match {
+      case Some(Main) =>
+        if(!AddressJourney.hasAnsweredUk(userAnswers) && !AddressJourney.hasAnsweredInternational(userAnswers)) {
+          Try(userAnswers)
+        } else {
+          userAnswers
+            .remove(IsNewAddressInUkPage)
+            .flatMap(_.remove(EnterYourUkAddressPage))
+            .flatMap(_.remove(WhatIsYourNewAddressUkPage))
+            .flatMap(_.remove(WhatIsYourNewAddressNonUkPage))
```
Comment (owen-opencast) [inline app/pages/answers/changeaddress/RemovePostalAddressPage.scala:43]: Think we also need a WhenDidYouStartLivingHere wiped here

**Review by owen-opencast — APPROVED** [review]

**Review by owen-opencast — APPROVED** [review]

**Comment by hmrc-web-operations** [comment]
Thank you for submitting a pull request on this HMRC repository. A member of the HMRC team will review your changes and action accordingly.

HMRC reviewer: please reference [this runbook](https://github.com/hmrc/build-and-deploy/blob/main/runbooks/reviewing_external_pull_requests.md) for guidance on how to proceed.

**Pull request merged**